internal/pkg/skuba/kubeadm/configmap.go

```diff
@@ -114,3 +117,44 @@ func RemoveAPIEndpointFromConfigMap(client clientset.Interface, node *v1.Node) e
 	}
 	return nil
 }
+
+// UpdateClusterConfigurationWithClusterVersion allows us to set certain configurations during init, but also during upgrades.
+// The configuration that we put here will be consistently set to newly created configurations, and when we upgrade a cluster.
+func UpdateClusterConfigurationWithClusterVersion(initCfg *kubeadmapi.InitConfiguration, clusterVersion *version.Version) {
+	setApiserverAdmissionPlugins(initCfg, clusterVersion)
+	setContainerImagesWithClusterVersion(initCfg, clusterVersion)
+}
+
+func setApiserverAdmissionPlugins(initCfg *kubeadmapi.InitConfiguration, clusterVersion *version.Version) {
+	if initCfg.APIServer.ControlPlaneComponent.ExtraArgs == nil {
+		initCfg.APIServer.ControlPlaneComponent.ExtraArgs = map[string]string{}
+	}
+	admissionPlugins := []string{}
+	if len(initCfg.APIServer.ExtraArgs["enable-admission-plugins"]) > 0 {
+		admissionPlugins = strings.Split(initCfg.APIServer.ExtraArgs["enable-admission-plugins"], ",")
+	}
+	// List of recommended plugins: https://git.io/JemEu
+	admissionPlugins = append(admissionPlugins,
```

Comment:
nit:

else?

I am not sure if it's fine to append to an existing list of admissionPlugins in the case you already have them.
Or maybe my brain farts, and I should totally go learn go.

Reply:
Can you please elaborate? not sure I'm following.

Reply:
@evrardjp This function call deletes duplicate admission plugins https://github.com/SUSE/skuba/blob/fa14c5c46877ffe4231909d3cfacc85588b03752/internal/pkg/skuba/kubeadm/configmap.go#L158

Reply:
shouldnt it be easier to have an `AppendNoDuplicates` so we only append if not in there?

Maybe its even more complicated of a func than the `UniqueStringSlice`, probably needs to iterate over the slice for each element, which `UniqueStringSlice` seems to iterate just once over the slice

Go its weird :)

Reply:
Discussed outside this, and it's basically the intended behaviour. Good then!